### DISCUSSION TITLE
External USB wifi adapter

### DISCUSSION THREAD

**Discussion by beursdummy** [jayofelony/pwnagotchi#105] [Q&A] [answered]
First of all hats off to the incredible work you've done so far on modifying the Pwnagotchi project!!

My question is if your build includes an automatic wlan switch when a USB adapter is found? I noticed that the Pwnagotchi was sending messages that wlan0 was off and it tried to restart it. Does that mean it also disables wlan0 and uses wlan1 when a USB adapter on wlan1 exists?

I'm experimenting with [pwnagotchi-external-wifi-plugin](https://github.com/unagisan69/pwnagotchi-external-wifi-plugin) but' I'm not sure if that conflicts with your build

**Comment by jayofelony** [comment]
No, the messages you see come from the fix_services plugin. The function of this plugin is to check whether the wifi firmware has crashed and to try and restart this. The firmware tends to crash some times, or go blind.

**Comment by beursdummy** [comment]
So would it make sense to ditch the plugin and create a logic that if wlan1mon is available I update the configvalue for iface to wlan1mon and reboot?

**Reply by jayofelony** [reply, marked as answer]
The only issue there is if your pi were to reboot, it randomly assignes wlan0 and wlan1 to an interface.
